forms/CopyInput/index.js

```diff
@@ -0,0 +1,131 @@
+'use strict'
+
+import React from 'react'
+import classnames from 'classnames'
+import bowser from 'bowser'
+import TextInput from '../TextInput'
+import Icon from '../../atoms/Icon'
+
+let copiedTimer
+
+export default React.createClass({
+  displayName: 'CopyInput',
+  propTypes: {
+    className: React.PropTypes.string,
+    disabled: React.PropTypes.bool,
+    label: React.PropTypes.string,
+    labelCopied: React.PropTypes.string,
+    labelCopy: React.PropTypes.string,
+    labelSelect: React.PropTypes.string,
+    layout: React.PropTypes.string,
+    name: React.PropTypes.string,
+    onBlur: React.PropTypes.func,
+    onKeyDown: React.PropTypes.func,
+    onTab: React.PropTypes.func,
+    spacing: React.PropTypes.string,
+    value: React.PropTypes.string
+  },
+
+  getDefaultProps() {
+    return {
+      disabled: false,
+      icon: null,
+      id: null,
+      label: 'Copy this value',
+      labelCopied: 'Copied',
+      labelCopy: 'Copy',
+      labelSelect: 'Select',
+      layout: 'full',
+      name: null,
+      onBlur() {},
+      onKeyDown() {},
+      onTab() {},
+      readOnly: true,
+      spacing: 'loose',
+      type: 'text',
+      value: ''
+    }
+  },
+
+  getInitialState() {
+    return { copied: false }
+  },
+
+  handleFocus({ inputElement }) {
+    if (inputElement && inputElement.setSelectionRange) {
+      inputElement.setSelectionRange(0, inputElement.value.length)
+    }
+  },
+
+  selectAndCopy() {
+    this.refs.copyInput.focus()
+
+    try {
+      document.execCommand('copy')
+      this.setState({ copied: true })
+
+      clearTimeout(copiedTimer)
+      copiedTimer = setTimeout(() => {
+        this.setState({ copied: false })
+      }, 4000)
+    } catch (error) {
+      console.warn('Copy command is not supported in this browser')
```

Comment:
I wonder if this error should appear underneath the input so that the user can actually know what went wrong. Not sure if this is worth holding the PR back